packages/core/saved-objects/core-saved-objects-server-internal/src/saved_objects_service.ts

```diff
@@ -194,7 +195,7 @@ export class SavedObjectsService
         if (this.started) {
           throw new Error('cannot call `registerType` after service startup.');
         }
-        this.typeRegistry.registerType(type);
+        this.typeRegistry.registerType(applyTypeDefaults(type));
```

Comment:
Doing that directly in the type registry class was too low level, just breaking too many tests. Also I think it make sense to have this done from the service instead.